gnuradio-runtime/lib/io_signature.cc

```diff
@@ -90,28 +90,43 @@ io_signature::io_signature(int min_streams,
       d_sizeof_stream_item(sizeof_stream_items),
       d_stream_buffer_type(buftypes)
 {
-    if (min_streams < 0 || (max_streams != IO_INFINITE && max_streams < min_streams))
-        throw std::invalid_argument("gr::io_signature(1)");
-
+    if (min_streams < 0) {
+        throw std::invalid_argument(
+            fmt::format("Minimum number of streams must be >= 0, is {:d}", min_streams));
+    }
+    if (max_streams != IO_INFINITE && max_streams < min_streams) {
+        throw std::invalid_argument(
+            fmt::format("Maximum number of streams ({:d}) needs to be at least as "
+                        "large as minimal number of streams ({:d})",
+                        max_streams,
+                        min_streams));
+    }
     if (sizeof_stream_items.empty()) {
-        throw std::invalid_argument("gr::io_signature(2)");
+        throw std::invalid_argument("Item size vector cannot be empty");
     }
-
-    for (size_t i = 0; i < sizeof_stream_items.size(); i++) {
-        if (max_streams != 0 && sizeof_stream_items[i] < 1)
-            throw std::invalid_argument("gr::io_signature(3)");
+    if (max_streams) {
+        decltype(sizeof_stream_items.size()) index = 0;
```

Comment:
Are there any cases where `size_t` wouldn't work here ... just to keep the syntax more readable?

Reply:
yeah, you're right!